lib/schematics/schematic.option.ts

```diff
@@ -1,13 +1,22 @@
-import { normalizeToKebabOrSnakeCase } from '../utils/formatting';
+import { normalizeToCase, formatString, CaseType } from '../utils/formatting';
+
+export type SchematicOptionConfig = {
+  caseType?: CaseType;
+};
+
+const defaultSchematicOptionConfig: SchematicOptionConfig = {
+  caseType: 'kebab-or-snake',
+};
 
 export class SchematicOption {
   constructor(
     private name: string,
     private value: boolean | string,
+    private schematicOptionConfig = defaultSchematicOptionConfig,
```

Comment:
Here I'm adding a `SchematicOptionConfig` to handle a set of generic options we could pass, including `CaseType`.

The first thing I tried was to add something like `SchmaticOption('caseNaming', 'camelCase')`, but I couldn't make it work because when we launch the command `nest generate controller emails --caseNaming=camel` we have two schematic options:
- `SchmaticOption('name', 'emails')`
- `SchmaticOption('suffix', 'controller')`

We need to apply the `camelCase` transform to both, hence why I used a 3 way option. 
I honestly don't like this approach very much, so I would love some feedback.

Reply:
Please resolve this, the approach changed.